sdk/python/foundation-models/system/inference/text-to-image/aacs-prepare-uai.ipynb

```diff
@@ -0,0 +1,434 @@
+    "    from azure.core.exceptions import ResourceExistsError\n",
+    "\n",
+    "    try:\n",
+    "        role_assignment_client.role_assignments.create(\n",
+    "            scope=scope,\n",
+    "            role_assignment_name=str(uuid.uuid4()),\n",
+    "            parameters=RoleAssignmentCreateParameters(\n",
+    "                role_definition_id=role_def.id,\n",
+    "                principal_id=uai_principal_id,\n",
+    "                principal_type=\"ServicePrincipal\",\n",
+    "            ),\n",
+    "        )\n",
+    "    except ResourceExistsError as ex:\n",
+    "        pass\n",
+    "    except Exception as ex:\n",
+    "        print(ex)\n",
+    "        raise ex\n",
+    "\n",
+    "    if role_name in uai_role_check_list:\n",
+    "        del uai_role_check_list[role_name]\n",
+    "    print(\n",
+    "        f\"Role assignment for {role_name} at the Azure AI Content Safety resource level completed.\"\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.4.3 Grant the user identity access to the Azure AI Content Safety (AACS) resource"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.mgmt.cognitiveservices import CognitiveServicesManagementClient\n",
+    "\n",
+    "aacs_client = CognitiveServicesManagementClient(credential, subscription_id)\n",
+    "\n",
+    "\n",
+    "def find_acs(accounts):\n",
+    "    return next(x for x in accounts if x.kind == \"ContentSafety\")\n",
+    "\n",
+    "\n",
+    "if aacs_name == \"\":\n",
+    "    for aacs_resource in aacs_client.accounts.list_by_resource_group(resource_group):\n",
+    "        assign_access_to_aacs(aacs_resource.id)\n",
+    "else:\n",
+    "    aacs_resource = aacs_client.accounts.get(resource_group, aacs_name)\n",
+    "    assign_access_to_aacs(aacs_resource.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.4.4 Assign AcrPull at the workspace container registry scope\n",
+    "Since we will create the Azure AI Content Safety (AACS) enabled text-to-image endpoint with User Assigned Identity, the user managed identity must have Storage Blob Data Reader permission on the storage account for the workspace, and AcrPull permission on the Azure Container Registry (ACR) for the workspace. Make sure your User Assigned Identity has the right permission."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "container_registry = workspace_resource.container_registry\n",
+    "\n",
+    "role_name = \"AcrPull\"\n",
+    "\n",
+    "role_defs = role_definition_client.role_definitions.list(scope=container_registry)\n",
+    "role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "from azure.core.exceptions import ResourceExistsError\n",
```

Comment:
we are using this inside assign_access_to_aacs() as well, should we import it just once above.